Data guide update

nollety opened the pull request:
# Description

Updates the data guide.

Most of the updating relates to #168 and affects the introduction page in particular.
All the other pages were updated by removing the outdated content warnings and the "Visualise the data" sections due to missing tutorials, and by referring to the introduction page about how to access the specific datasets.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

leroyvn commented on docs/rst/user_guide/data/intro.rst:
```diff
@@ -23,37 +21,42 @@ data.
+This global data store aggregates multiple subordinated data stores based on
+the size and maturity level of data files they manage.
+List a data store' registered data sets by reading its ``registry`` property,
+i.e.:
```
```suggestion
*i.e.*:
```

leroyvn commented on docs/rst/user_guide/data/radprops.rst:
```diff
@@ -3,17 +3,15 @@
 Atmosphere radiative properties data sets are created by
-:class:`~eradiate.radprops.rad_profile.RadProfile` objects'
-:meth:`~eradiate.radprops.rad_profile.RadProfile.to_dataset` method.
+:class:`~eradiate.radprops._core.RadProfile` objects'
+:meth:`~eradiate.radprops._core.RadProfile.eval_dataset` method.
```
I'd expect that not to render correctly: `RadProfile` is referenced as `eradiate.radprops.RadProfile`. Anyway, no need to bother, I think the following should work:
```suggestion
:meth:`.RadProfile.eval_dataset` method.
```

leroyvn commented on docs/rst/user_guide/data/intro.rst:
```diff
@@ -23,37 +21,42 @@ data.
 .. code-block:: python
 
-   ds = eradiate.data.open("path/to/my/data.nc")
+   list(data_store.stores["large_files_stable"].registry)
```
Quite inconvenient IMO: this is typically something which should be implemented as a feature of the `eradiate data info` command.

nollety replied:
I agree

nollety replied:
I created eradiate/eradiate-issues#149 to address that

leroyvn reviewed (changes requested):
Hi @nollety, thank you for checking this out. I made a few suggestions.

nollety commented:
Hi @leroyvn, thanks for your feedback, I applied your suggestions, squashed the commits and rebased on latest main. I also created eradiate/eradiate-issues#149 to address your comment about listing available data sets.

leroyvn commented:
Thanks @nollety, let's merge this.
